lib/src/components/NxTransferList/NxTransferList.tsx

```diff
@@ -65,7 +64,14 @@ export default function NxTransferList
   const availableCount = allItems.length - selectedItemsArray.length,
       selectedCount = selectedItemsArray.length;
 
-  const handleOnChangeProp = (array: T[]) => onChangeProp((allowReordering ? array : new Set(array)) as P);
+  const handleOnChangeProp = (array: T[]) => {
+    if (allowReordering) {
+      (onChangeProp as (newSelected: T[]) => void)(array);
+    }
+    else {
+      (onChangeProp as (newSelected: Set<T>) => void)(new Set(array));
+    }
+  };
```

Comment:
I had hoped this would get rid of the cast here but instead it made it worse.  It seems typescript is not able to fully track the ramifications of the `if (allowReordering)` guard here.  I'm not sure if that's because we're nested pretty far down into the component here, or if it's something about the way react's props work.

Reply:
> I had hoped this would get rid of the cast here but instead it made it worse. It seems typescript is not able to fully track the ramifications of the `if (allowReordering)` guard here. I'm not sure if that's because we're nested pretty far down into the component here, or if it's something about the way react's props work.

Hmm yeah :(. I tried something similar, but got stuck at `Argument of type 'Set<T> | T[]' is not assignable to parameter of type 'Set<T> & T[]'`